inference-engine/src/plugin_api/cpp_interfaces/impl/ie_infer_async_request_thread_safe_default.hpp

```diff
@@ -378,6 +446,7 @@ class AsyncInferRequestThreadSafeDefault : public AsyncInferRequestThreadSafeInt
         }, std::move(callbackExecutor));
     }
 
+    std::atomic_bool _isRequestBusy = {false};
```

Comment:
Just out of curiosity, why we use `std::atomic` for synchronization instead of `std::mutex` and for example `std::unique_lock( mutex_type& m, std::try_to_lock_t t )` ?

Reply:
Seems it was the most simple way to implement declared logic before. 

BTW This merge request is intended as first step of  base synchronization logic optimization and beatification. I have no vision how it can be done yet. So if you have any thoughts you can provide separate  pull request and discuss the solution. 

I think that there are too many synchronization primitives today and your suggestion with single mutex can be the right way to solve the issue.